site/components/NavLinkButton.vue

```diff
@@ -0,0 +1,18 @@
+<template>
+    <a class="no-animation btn btn-primary w-full">
+        <div class="flex flex-row items-center w-full">
+            <font-awesome-icon :icon="$props.linkIcon" class="text-lg" />
+            <span class='ml-4 font-semibold'>{{ $props.title }}</span>
+        </div>
+    </a>
+</template>
+
+<script setup lang="ts">
+import type { IconDefinition } from '@fortawesome/fontawesome-svg-core';
+
+
+defineProps<{
+    title: String,
```

Comment:
This is the only non-suggestion, as it infects the typesystem.

```suggestion
    title: string,
```
Avoid `String`, use `string`

[Some documentation for this, as its just one of those things](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#number-string-boolean-symbol-and-object)

Reply:
Avoid `String`, use `string`

[Some documentation for this, as its just one of those things](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#number-string-boolean-symbol-and-object)